cli-plugins/manager/hooks.go

```diff
@@ -0,0 +1,180 @@
+//
+// when being executed after the command:
+// `docker pull alpine`
+// will result in the message:
+// "Run this image with `docker run alpine`"
+func TemplateReplaceArg(i int) string {
+	return fmt.Sprintf(hookTemplateArg, strconv.Itoa(i))
+}
+
+// RunPluginHooks calls the hook subcommand for all present
+// CLI plugins that declare support for hooks in their metadata
+// and parses/prints their responses
+func RunPluginHooks(dockerCli command.Cli, rootCmd, subCommand *cobra.Command) error {
+	plugins, err := ListPlugins(dockerCli, rootCmd)
+	if err != nil {
+		return err
+	}
+	subCmdName := subCommand.Name()
+
+	hookMessages := make([]string, 0, len(plugins))
+	for _, p := range plugins {
+		p := p
+		if !p.supportsHookCommand(subCmdName) {
+			continue
+		}
+
+		hookReturn, err := p.RunHook(subCmdName)
+		// Skip misbehaving plugins, but don't halt execution
+		if err != nil {
+			continue
+		}
+		hookMessages = append(hookMessages, parseHooks(hookReturn.Template, subCommand))
+	}
+
+	printHookMessages(dockerCli.Err(), hookMessages)
+	return nil
+}
+
+const (
+	hookTemplateCommandName = "{docker-cli-plugin-hook-cmd-name}"
+	hookTemplateFlagValue   = "{docker-cli-plugin-hook-flag-%s}"
+	hookTemplateArg         = "{docker-cli-plugin-hook-arg-%s}"
+)
+
+func parseHooks(template string, cmd *cobra.Command) string {
+	ret := template
+	for {
+		parsed, foundMatch := parseFlagValue(ret, cmd)
+		ret = parsed
+		if !foundMatch {
+			break
+		}
+	}
+	for {
+		parsed, foundMatch := parseArg(ret, cmd)
+		ret = parsed
+		if !foundMatch {
+			break
+		}
+	}
+	return strings.ReplaceAll(ret, hookTemplateCommandName, cmd.Name())
+}
+
+var (
+	flagRegex = regexp.MustCompile(fmt.Sprintf(hookTemplateFlagValue, `([\w-]+)`))
+	argRegex  = regexp.MustCompile(fmt.Sprintf(hookTemplateArg, `([\w-]+)`))
+)
+
+func parseFlagValue(template string, cmd *cobra.Command) (string, bool) {
+	matches := flagRegex.FindStringSubmatch(template)
+	if len(matches) == 0 {
+		return template, false
+	}
+	for _, m := range matches[1:] {
+		flag := cmd.Flag(m)
+		if flag == nil {
+			continue
+		}
+		value := flag.Value.String()
+		template = strings.ReplaceAll(template, TemplateReplaceFlagValue(m), value)
+	}
+	return template, true
+}
+
+func parseArg(template string, cmd *cobra.Command) (string, bool) {
+	matches := argRegex.FindStringSubmatch(template)
+	if len(matches) == 0 {
+		return template, false
+	}
+	for _, m := range matches[1:] {
+		i, err := strconv.Atoi(m)
+		if err != nil {
+			continue
+		}
+		if i >= len(cmd.Flags().Args()) {
+			return template, false
+		}
+		template = strings.ReplaceAll(template, TemplateReplaceArg(i), cmd.Flags().Arg(i))
+	}
+	return template, true
+}
+
+func printHookMessages(out io.Writer, messages []string) {
+	if len(messages) == 0 {
+		return
+	}
+
+	fmt.Fprintln(out, aec.LightBlueF.Apply("\nNext steps:"))
+	for _, m := range messages {
+		fmt.Fprintf(out, "    - %s\n", m)
```

Comment:
Looks like "list" layout is enforced here. Should we also make sure that each message is 1-line only?

Reply:
Maybe we could format each line of a message (after the first) so they at least start with the same indentaton? (a small terminal would wrap the lines, so in that case it'd look ugly anyway)

Reply:
> Maybe we could format each line of a message (after the first) so they at least start with the same indentaton?
```
Next steps:
    - Keep in mind that lorem ipsum dolor sit amet.
      (also, pellentesque ac eleifend tellus, ac malesuada dolor)
    - Nullam luctus rutrum fringilla
```
Like this? Not sure if this is clearer or not for me.

Reply:
Exactly! I think it looks much better (but that might just be a strong bias on my side 😅)

Reply:
Today in `docker scout` we don't display them as list items, even if there's multiple items:
<img width="957" alt="image" src="https://github.com/docker/cli/assets/1542321/76df344c-f969-4219-b8c5-07e36b1d307a">

Reply:
I'll take a jab at it – I like the idea, but slightly concerned about totally breaking formatting with the indentation in smaller terminals.